cloud/google/gce_img.py

```diff
@@ -0,0 +1,174 @@
+try:
+  from libcloud.compute.types import Provider
+  from libcloud.compute.providers import get_driver
+  from libcloud.common.google import GoogleBaseError
+  from libcloud.common.google import ResourceNotFoundError
+  _ = Provider.GCE
+except ImportError:
+  print('failed=True '
+        "msg='libcloud with GCE support is required for this module.'")
+  sys.exit(1)
+
+DOCUMENTATION = '''
+---
+module: gce_img
+short_description: utilize GCE image resources
+description:
+    - This module can create and delete GCE private images from gzipped
+      compressed tarball containing raw disk data or from existing detached
+      disks in any zone. U(https://cloud.google.com/compute/docs/images)
+options:
+  name:
+    description:
+      - the name of the image to create
+    required: true
+    default: null
+    aliases: []
+  source:
+    description:
+      - the source disk or the Google Cloud Storage URI to create the image from
+    required: false
+    default: null
+    aliases: []
+  state:
+    description:
+      - desired state of the image
+    required: false
+    default: "present"
+    choices: ["active", "present", "absent", "deleted"]
+    aliases: []
+  zone:
+    description:
+      - the zone of the disk specified by source
+    required: false
+    default: "us-central1-a"
+    aliases: []
+  service_account_email:
+    version_added: "1.6"
+    description:
+      - service account email
+    required: false
+    default: null
+    aliases: []
+  pem_file:
+    version_added: "1.6"
+    description:
+      - path to the pem file associated with the service account email
+    required: false
+    default: null
+    aliases: []
+  project_id:
+    version_added: "1.6"
+    description:
+      - your GCE project ID
+    required: false
+    default: null
+    aliases: []
+
+requirements: [ "libcloud" ]
+author: Peter Tan <ptan@google.com>
+'''
+
+EXAMPLES = '''
+# Create an image named test-image from the disk 'test-disk' in zone us-central1-a.
+- gce_img:
+    name: test-image
+    source: test-disk
+    zone: us-central1-a
+    state: present
+
+# Delete an image named test-image in zone us-central1-a.
+- gce_img:
+    name: test-image
+    zone: us-central1-a
+    state: deleted
+'''
+
+
+def main():
+  module = AnsibleModule(
+      argument_spec=dict(
+          name=dict(required=True),
+          source=dict(),
+          state=dict(default='present'),
+          zone=dict(default='us-central1-a'),
+          service_account_email=dict(),
+          pem_file=dict(),
+          project_id=dict(),
+      )
+  )
+
+  gce = gce_connect(module)
+
+  name = module.params.get('name')
+  source = module.params.get('source')
+  state = module.params.get('state')
+  zone = module.params.get('zone')
+  changed = False
+
+  try:
+    image = gce.ex_get_image(name)
+  except GoogleBaseError, e:
+    module.fail_json(msg=str(e), changed=False)
+
+  # user wants to create an image.
+  if state in ['active', 'present'] and not image:
```

Comment:
Along with adding `choices` to the `state` argument, you can simplify this logic, as well as similar logic to this farther down.